x-pack/plugins/observability_solution/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.test.ts

```diff
@@ -2366,8 +2389,7 @@ describe('The metric threshold rule type', () => {
         group: id,
         reason,
         timestamp: mockNow.toISOString(),
-        viewInAppUrl: 'http://localhost:5601/app/metrics/explorer',
-
+        viewInAppUrl: undefined,
```

Comment:
@crespocarlos Is this the correct case now if we don't have a view or fallback url?

Reply:
After fixing this,`rule tags and source tags are combined in alert context` test will fail. We need to fix the `getMetricsViewInAppUrl` to also redirect to `metricsExplorerLocator` in case `fields[findInventoryModel(supportedAssetType).fields.id]` is `undefined`